db/workflow.go

```diff
@@ -108,7 +108,7 @@ func insertActionList(ctx context.Context, db *sql.DB, yamlData string, id uuid.
 		}
 
 		workerID, err := getWorkerID(ctx, db, task.WorkerAddr)
-		if err != nil {
+		if err != nil && err != sql.ErrNoRows {
```

Comment:
Can you clear up why its ok to not return an error when err == sql.ErrNoRows, its not immediately obvious to me.
On second thought why don't we return an error in that case?

Reply:
oh sorry it's because `workerID == ""` is basically the same as `sql.ErrNoRows` in this case. But  I realize it's a better idea to switch the order of the check so it checks for ErrNoRows first and then any other error.

Reply:
ah yes I see. Yeah lets handle ErrNoRows first&explicitly so its clearer :D :+1: